### PR TITLE
Update publications component to match refined mocks. #86.

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers

#86 

@MillenniumFalconMechanic 

### Changes
- added publications component.
- added publications story and publications story to project story.
- added publications to `projectTransformer` and `projectViewModelBuilder`.
- updated project entities with project related interfaces.

### CODE REVIEW

File: explorer/app/components/Project/components/Publications/publications.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { Typography } from "@mui/material";
+import React from "react";
+
+// App dependencies
+import { ANCHOR_TARGET, Link } from "../../../Links/components/Link/link";
+import { SectionDetailsEmpty } from "../Section/components/SectionDetailsEmpty/sectionDetailsEmpty";
+import { Section } from "../Section/section";
+
+export interface Publication {
+  doi?: string;
```
Comment (frano-m) [inline explorer/app/components/Project/components/Publications/publications.tsx:11]: Would you mind please reviewing the `Publication` interface carefully - the API appears to always have `doi`, and `officialHcaPublication` (n.b. boolean and also null values observed), however the "updated project" ENTRIES model is optional `doi` and `officialHcaPublication`... Should I type the API separately?

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 2]
```diff
@@ -283,3 +314,33 @@ function isValidUrl(testUrl: string): boolean {
...
+    updatedProject &&
+    updatedProject.publications &&
+    updatedProject.publications.length > 0
+  ) {
+    return updatedProject.publications;
+  }
+  return publicationsResponse.map((publicationResponse) => {
+    return {
+      ...publicationResponse,
+      doi: publicationResponse.doi || "", // Maps any null value to string.
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/common/projectTransformer.ts:341]:
```suggestion
      doi: publicationResponse.doi ?? "", // Maps any null value to string.
```

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 3]
```diff
@@ -283,3 +314,33 @@ function isValidUrl(testUrl: string): boolean {
...
+      officialHcaPublication:
+        publicationResponse.officialHcaPublication || false, // Maps any null value to "false".
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/common/projectTransformer.ts:343]:
```suggestion
        publicationResponse.officialHcaPublication ?? false, // Maps any null value to "false".
```

**Comment by frano-m** [comment]
Thank you so much @MillenniumFalconMechanic ❤️

**Pull request merged**